src/fromager/commands/build.py

```diff
@@ -165,13 +175,25 @@ def _build(
 
 def _is_wheel_built(
     wkctx: context.WorkContext, dist_name: str, dist_version: Version
-) -> bool:
+) -> tuple[bool, int]:
     req = Requirement(f"{dist_name}=={dist_version}")
 
     try:
         logger.info(f"{req.name}: checking if {req} was already built")
-        sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
-        return True
+        url, _ = sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
+        build_tag = wkctx.settings.build_tag(req.name, dist_version)
+        wheel_filename = urlparse(url).path.rpartition("/")[-1]
+        _, _, build_tag_from_name, _ = parse_wheel_filename(wheel_filename)
+        if build_tag_from_name:
+            increment_by = max(build_tag_from_name[0] - build_tag, 0)
+            if increment_by > 0:
+                increment_by += 1  # make the new build tag greater than the latest one
+                logger.warn(
+                    f"{dist_name}: changelog for version {dist_version} is inconsistent. Found build tag {build_tag_from_name[0]} but expected {build_tag}. Will tag the wheel with build number {build_tag + increment_by}"
```

Comment:
This is to solve this problem: https://github.com/python-wheel-build/fromager/issues/318#issuecomment-2287160456

We will warn about the changelog and then use the latest build tag + 1 to build the wheel